src/main/java/groots/sad/project/MassTransitSystem/manager/SystemEfficiencyCalculator.java

```diff
@@ -1,9 +1,58 @@
 package groots.sad.project.MassTransitSystem.manager;
 
+import groots.sad.project.MassTransitSystem.entity.Bus;
+import groots.sad.project.MassTransitSystem.entity.BusStop;
+
+import java.util.List;
+
+/**
+ * The class is singleton class which will calculate the efficiency of the system.
+ * The object instantiation is an eager instantiation to avoid concurrency issues.
+ */
 public class SystemEfficiencyCalculator {
 
+    private double kSpeed;
+    private double kCapacity;
+    private double kWaiting;
```

Comment:
Shouldn't kwaiting and kcombined be public ?

Reply:
Actually, we can write getters for them if they are used in different class but making them public would then make them available to other classes and our motto of data hiding will not be achieved .